rosdoc2/verbs/build/builders/sphinx_builder.py

```diff
@@ -320,7 +334,9 @@ def __init__(self, builder_name, builder_entry_dictionary, build_context):
 
     def build(self, *, doc_build_folder, output_staging_directory):
         # Check that doxygen_xml_directory exists relative to output staging, if specified.
-        if self.doxygen_xml_directory is not None:
+        if (self.doxygen_xml_directory is not None and
+                (self.build_context.build_type in ('ament_cmake', 'cmake') or
+                 self.build_context.always_run_doxygen)):
             self.doxygen_xml_directory = \
                 os.path.join(output_staging_directory, self.doxygen_xml_directory)
```

Comment:
nitpick: similar suggestion to make it clearer to read:

```suggestion
        if (
            self.doxygen_xml_directory is not None and
            (
                self.build_context.build_type in ('ament_cmake', 'cmake') or
                self.build_context.always_run_doxygen
            )
        ):
            self.doxygen_xml_directory = \
                os.path.join(output_staging_directory, self.doxygen_xml_directory)
```

This one also avoids indentations that are not a multi of four spaces.

Reply:
Also, don't be afraid to use more variables for intermediate values, e.g. the second half of the condition could be `should_run_doxygen` which is calculated on a previous line.

Reply:
Addressed in 1d7e75d.